### PR TITLE
output/httpd: reject some well-known request paths

### PR DESCRIPTION
**Pull request by cathugger**
Return `404 not found` for some common well-known paths, as clients requesting them usually do that automatically and don't expect endless audio stram.
Also use strncmp instead of memcmp, as memcmp use may result in out of bounds access.

### CODE REVIEW

File: src/output/plugins/httpd/HttpdClient.cxx [hunk 1]
```diff
@@ -71,10 +71,10 @@ HttpdClient::HandleLine(const char *line) noexcept
 	assert(state != State::RESPONSE);
 
 	if (state == State::REQUEST) {
-		if (memcmp(line, "HEAD /", 6) == 0) {
+		if (strncmp(line, "HEAD /", 6) == 0) {
```
Comment (MaxKellermann) [inline src/output/plugins/httpd/HttpdClient.cxx]: What does this have to do with rejecting well-known paths?

File: src/output/plugins/httpd/HttpdClient.cxx [hunk 2]
```diff
@@ -129,14 +141,21 @@ HttpdClient::SendResponse() noexcept
...
+			"404 not found";
+	} else if (metadata_requested) {
 		allocated =
 			icy_server_metadata_header(httpd.name, httpd.genre,
 						   httpd.website,
 						   httpd.content_type,
 						   metaint);
 		response = allocated.c_str();
-       } else { /* revert to a normal HTTP request */
+	} else { /* revert to a normal HTTP request */
```
Comment (MaxKellermann) [inline src/output/plugins/httpd/HttpdClient.cxx]: Why a whitespace change here? What does this have to do with rejecting certain paths?

**Comment by MaxKellermann** [comment]
The commit is bad because it mixes more than one change into one commit.

**Comment by MaxKellermann** [comment]
f9ca2f52c17aa037c5cffb28b9052a4003cad36d

**Comment by cathugger** [comment]
Oh, sorry, I guess it is, I just fixed whatever else bothered me while I was at it, but I guess putting that into single commit is worse for anyone later wanting to make sense of them.
Thanks for splitting it so I didn't need to.
